src/platform/platform_misc.c

```diff
@@ -68,29 +69,61 @@ static int windows_system(const char *cmd)
 {
     PROCESS_INFORMATION p_info;
     STARTUPINFOW s_info;
-    DWORD ReturnValue;
+    HANDLE hp, p_stderr;
+    DWORD handleInfo, flags, ret, err = ERROR_SUCCESS;
 
     memset(&s_info, 0, sizeof(s_info));
     memset(&p_info, 0, sizeof(p_info));
     s_info.cb = sizeof(s_info);
+    s_info.dwFlags = STARTF_USESTDHANDLES;
 
-    char_os *utf16cmd;
-    utf16cmd = caml_stat_strdup_to_os(cmd);
-    if (CreateProcessW(NULL, utf16cmd, NULL, NULL, FALSE, CREATE_NO_WINDOW, NULL, NULL, &s_info, &p_info))
-    {
+    s_info.hStdInput = INVALID_HANDLE_VALUE;
+    /* If needed, duplicate stderr to make sure it is inheritable */
+    p_stderr = GetStdHandle(STD_ERROR_HANDLE);
+    if (p_stderr == INVALID_HANDLE_VALUE) {
+        err = GetLastError(); goto ret1;
+    }
+    if (! GetHandleInformation(p_stderr, &handleInfo)) {
+        err = GetLastError(); goto ret1;
+    }
+    if (! (handleInfo & HANDLE_FLAG_INHERIT)) {
+        hp = GetCurrentProcess();
+        if (! DuplicateHandle(hp, p_stderr, hp, &(s_info.hStdError),
+                              0, TRUE, DUPLICATE_SAME_ACCESS)) {
```

Comment:
Is this call to `DuplicateHandle` necessary, given that the following call to `DuplicateHandle` below also makes the `stderr` handle inheritable?

Reply:
We're redirecting stdout to stderr so I'm considering stderr first. We need to make sure stderr of the parent (`p_stderr`) is inheritable (keep-on-exec), so if it's not, I duplicate it to mark in inheritable. If it is already inheritable, no need to duplicate it.
I think you're right that a `DuplicateHandle` is superfluous, but I'd say it's the second. It may be possible to set `s_info.hStdOutput = s_info.hStdError` instead of duplicating the handle again.